core/AutocorrectSuggestion.cc

```diff
@@ -27,26 +27,32 @@ bool hasSeen(const UnorderedSet<Loc> &seen, Loc loc) {
 
 UnorderedMap<FileRef, string> AutocorrectSuggestion::apply(vector<AutocorrectSuggestion> autocorrects,
                                                            UnorderedMap<FileRef, string> sources) {
+    vector<pair<core::Loc, string>> edits;
+    for (auto &autocorrect : autocorrects) {
+        move(autocorrect.edits.begin(), autocorrect.edits.end(), back_inserter(edits));
+    }
+
     // Sort the locs backwards
-    auto compare = [](const AutocorrectSuggestion &left, const AutocorrectSuggestion &right) {
-        if (left.loc.file() != right.loc.file()) {
-            return left.loc.file().id() > right.loc.file().id();
+    auto compare = [](const pair<core::Loc, string> &left, const pair<core::Loc, string> &right) {
+        if (left.first.file() != right.first.file()) {
+            return left.first.file().id() > right.first.file().id();
         }
 
-        auto a = left.loc.beginPos();
-        auto b = right.loc.beginPos();
+        auto a = left.first.beginPos();
+        auto b = right.first.beginPos();
         if (a != b) {
             return a > b;
         }
 
         return false;
     };
-    fast_sort(autocorrects, compare);
+    fast_sort(edits, compare);
 
     UnorderedSet<Loc> seen; // used to make sure nothing overlaps
     UnorderedMap<FileRef, string> ret;
-    for (auto &autocorrect : autocorrects) {
-        auto &loc = autocorrect.loc;
+    for (auto &edit : edits) {
```

Comment:
`const auto &[loc, replacement]: edits`

Reply:
Ah, I grepped for `auto [...` to see if we did this anywhere else but should have grepped with the `&`.

Reply:
`auto [` makes a copy and you rarely want that